test/uk/gov/hmrc/tai/controllers/income/IncomeControllerSpec.scala

```diff
@@ -191,15 +200,452 @@ class IncomeControllerSpec extends PlaySpec
+      val mockIncomeService = mock[IncomeService]
+      when(mockIncomeService.taxCodeIncomes(any(), Matchers.eq(TaxYear().next))(any()))
+        .thenReturn(Future.successful(taxCodeIncomes))
+
+      val mockEmploymentService = mock[EmploymentService]
+      when(mockEmploymentService.employments(any[Nino], any[TaxYear])(any[HeaderCarrier]))
+        .thenReturn(Future.successful(Seq(employment)))
+
+      val sut = createSUT(incomeService = mockIncomeService, employmentService = mockEmploymentService, authentication = loggedInAuthenticationPredicate)
+      val result = sut.matchedTaxCodeIncomesForYear(nino, TaxYear().next, "BananaIncome", "Live")(FakeRequest())
+
+      status(result) mustBe OK
+      contentAsJson(result) mustBe expectedJsonEmpty
+    }
+
+    "return NotAuthorized when the user is not logged in" in {
+      val sut = createSUT(authentication = notLoggedInAuthenticationPredicate)
+      val result = sut.matchedTaxCodeIncomesForYear(nino, TaxYear().next, "EmploymentIncome", "Live")(FakeRequest())
+      ScalaFutures.whenReady(result.failed) { e =>
+        e mustBe a[MissingBearerToken]
+      }
+    }
+
+    "return NotFound when a NotFoundException occurs" in {
+      val mockIncomeService = mock[IncomeService]
+      when(mockIncomeService.taxCodeIncomes(any(), Matchers.eq(TaxYear().next))(any()))
+        .thenReturn(Future.failed(new NotFoundException("Error")))
+
+      val SUT = createSUT(mockIncomeService)
+      val result = SUT.matchedTaxCodeIncomesForYear(nino, TaxYear().next, "EmploymentIncome", "Live")(FakeRequest())
+
+      status(result) mustBe NOT_FOUND
+    }
+
+    "return BadRequest when a BadRequestException occurs" in {
+      val mockIncomeService = mock[IncomeService]
+      when(mockIncomeService.taxCodeIncomes(any(), Matchers.eq(TaxYear().next))(any()))
+        .thenReturn(Future.failed(new BadRequestException("Error")))
+
+      val SUT = createSUT(mockIncomeService)
+      val result = SUT.matchedTaxCodeIncomesForYear(nino, TaxYear().next, "EmploymentIncome", "Live")(FakeRequest())
+
+      status(result) mustBe BAD_REQUEST
+    }
+  }
+
+  "nonMatchingCeasedEmployments" must {
+
+    val taxCodeIncomes = Seq(TaxCodeIncome(PensionIncome, Some(1), BigDecimal(1100),
+      "PensionIncome", "1150L", "Employer1", Week1Month1BasisOperation, Live, BigDecimal(0), BigDecimal(0), BigDecimal(0)),
+      TaxCodeIncome(EmploymentIncome, Some(2), BigDecimal(0),
+        "EmploymentIncome", "1100L", "Employer2", OtherBasisOperation, Ceased, BigDecimal(321.12), BigDecimal(0), BigDecimal(0)),
+      TaxCodeIncome(EmploymentIncome, Some(3), BigDecimal(0),
+        "EmploymentIncome", "1100L", "Employer2", OtherBasisOperation, Live, BigDecimal(321.12), BigDecimal(0), BigDecimal(0)))
+
+    val employment = Employment("company name", Some("888"), new LocalDate(TaxYear().next.year, 5, 26),
+      None, Nil, "", "", 2, Some(100), hasPayrolledBenefit = false, receivingOccupationalPension = true)
+
```

Comment:
Is this the same as well? The naming of variables for specific employments might help here

Reply:
the same as what? I think this is fine